pkgs/build-support/rust/carnix.nix

```diff
@@ -0,0 +1,537 @@
+# This file contains the nix expression necessary to compile carnix, a
+# tool to generate nix expression from Rust Cargo.lock files.
+#
+# Starts with functions that output crates, and then calls them with
+# the proper dependencies computed from the carnix's own Cargo.lock.
+
+{ pkgs, ... }:
+with pkgs;
+let release = true;
+    verbose = true;
+    aho_corasick_0_6_3_ = { dependencies?[], buildDependencies?[], features?[] }: buildRustCrate {
```

Comment:
Could the arguments and inherit stuff be wrapped up into a helper function to reduce duplication here, and release-ness made dynamic? For example:
```nix
let verbose = true;
    helper = { ... } @ args: { dependencies ? [], buildDependencies ? [], features ? [], release ? true }:
               buildRustCrate (args // { inherit dependencies buildDependencies features release verbose; });
    aho_corasick_0_6_3_ = helper { crateName = "aho-corasick"; etc };
```